src/main/kotlin/com/github/mitchellolsthoorn/testgenie/evosuite/Runner.kt

```diff
@@ -0,0 +1,133 @@
+
+            val testResultName = "test_gen_result_$ts"
+            val serializeResultPath = "\"$testResultDirectory$testResultName\""
+
+            val settingsState = TestGenieSettingsService.getInstance().state
+
+            val command = SettingsArguments(projectClassPath, projectPath, serializeResultPath, classFQN).build()
+
+            if (!settingsState?.seed.isNullOrBlank()) command.add("-seed=${settingsState?.seed}")
+            if (!settingsState?.configurationId.isNullOrBlank()) command.add("-Dconfiguration_id=${settingsState?.configurationId}")
+
+            val cmd = ArrayList<String>()
+            cmd.add(javaPath)
+            cmd.add("-jar")
+            cmd.add(evoSuitePath)
+            cmd.addAll(command)
+
+            val cmdString = cmd.fold(String()) { acc, e -> acc.plus(e).plus(" ") }
+
+            log.info("Starting EvoSuite with arguments: $cmdString")
+            log.info("Results will be saved to $serializeResultPath")
+
+            runEvoSuite(cmd, projectPath)
+
+            return testResultName
+
+        }
+
+        fun runEvoSuiteForMethod(
+            projectPath: String, projectClassPath: String, classFQN: String, method: String
+        ): String {
+
+            val javaPath = "java"// TODO: Source from config
+
+            val pluginsPath = System.getProperty("idea.plugins.path")
+            val evoSuitePath = "$pluginsPath/TestGenie/lib/evosuite.jar"
+
+            val ts = System.currentTimeMillis()
+            val sep = File.separatorChar
+            val testResultDirectory = "${FileUtilRt.getTempDirectory()}${sep}testGenieResults${sep}"
+
+            val testResultName = "test_gen_result_$ts"
+            val serializeResultPath = "\"$testResultDirectory$testResultName\""
+
+            val settingsState = TestGenieSettingsService.getInstance().state
+
+            val command =
+                SettingsArguments(projectClassPath, projectPath, serializeResultPath, classFQN).forMethodPrefix(method)
+                    .build()
+
+            if (!settingsState?.seed.isNullOrBlank()) command.add("-seed=${settingsState?.seed}")
+            if (!settingsState?.configurationId.isNullOrBlank()) command.add("-Dconfiguration_id=${settingsState?.configurationId}")
+
+            val cmd = ArrayList<String>()
+            cmd.add(javaPath)
+            cmd.add("-jar")
+            cmd.add(evoSuitePath)
+            cmd.addAll(command)
+
+            val cmdString = cmd.fold(String()) { acc, e -> acc.plus(e).plus(" ") }
+
+            log.info("Starting EvoSuite with arguments: $cmdString")
+            log.info("Results will be saved to $serializeResultPath")
+
+            runEvoSuite(cmd, projectPath)
+
+            return testResultName
+        }
+
+
+        private fun runEvoSuite(cmd: ArrayList<String>, workDir: String) {
+            val evoSuiteProcessTimeout: Long = 12000000 // TODO: Source from config
+
+            Thread {
```

Comment:
A tip for next sprint is to refactor this into a backgroundable task, so that users know that EvoSuite is running in the background :)

Reply:
Thanks! Good idea. We will indeed address it in the next sprint, as it has also been specified by the client